### PR TITLE
FiBo-172: spying possibility for password fields

### PR DESCRIPTION
**Pull request by XSnelliusX**

### CODE REVIEW

File: android-app/app/src/main/res/layout/create_account.xml [hunk 1]
```diff
@@ -34,7 +33,7 @@
         <com.google.android.material.textfield.TextInputEditText
             android:id="@+id/create_account_email"
             android:layout_width="match_parent"
-            android:layout_height="48dp"
+            android:layout_height="match_parent"
```
Comment (Jonas-Karl) [inline android-app/app/src/main/res/layout/create_account.xml:36]:
```suggestion
            android:layout_height="wrap_content"
```

File: android-app/app/src/main/res/layout/activity_login.xml [hunk 2]
```diff
@@ -42,14 +42,16 @@
...
+        app:endIconContentDescription="password_toggle"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toBottomOf="@id/login_email_layer">
 
         <com.google.android.material.textfield.TextInputEditText
             android:id="@+id/login_password"
             android:layout_width="match_parent"
-            android:layout_height="48dp"
+            android:layout_height="match_parent"
```
Comment (Jonas-Karl) [inline android-app/app/src/main/res/layout/activity_login.xml:54]:
```suggestion
            android:layout_height="wrap_content"
```

File: android-app/app/src/main/res/layout/activity_login.xml [hunk 3]
```diff
@@ -31,7 +31,7 @@
         <com.google.android.material.textfield.TextInputEditText
             android:id="@+id/login_email"
             android:layout_width="match_parent"
-            android:layout_height="48dp"
+            android:layout_height="match_parent"
```
Comment (Jonas-Karl) [inline android-app/app/src/main/res/layout/activity_login.xml:34]:
```suggestion
            android:layout_height="wrap_content"
```

File: android-app/app/src/main/res/layout/create_account.xml [hunk 4]
```diff
@@ -45,14 +44,16 @@
...
+        app:endIconContentDescription="password_toggle"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toBottomOf="@id/create_account_email_layer">
 
         <com.google.android.material.textfield.TextInputEditText
             android:id="@+id/create_account_password"
             android:layout_width="match_parent"
-            android:layout_height="48dp"
+            android:layout_height="match_parent"
```
Comment (Jonas-Karl) [inline android-app/app/src/main/res/layout/create_account.xml:56]:
```suggestion
            android:layout_height="wrap_content"
```

File: android-app/app/src/androidTest/java/de/dhbw/ka/se/fibo/login/LoginActivityTest.java [hunk 5]
```diff
@@ -170,10 +164,29 @@ public void testValidInput() {
...
-    }
+        // click on the visibility toggle
+        onView(withContentDescription("password_toggle"))
+                .perform(click());
 
+        // tests that the password is readable
+        activityScenarioRule.getScenario().onActivity(activity -> {
+            EditText passwordFieldText = activity.findViewById(R.id.login_password);
+            assertEquals("testPassword", passwordFieldText.getLayout().getText().toString());
+        });
```
Comment (Jonas-Karl) [inline android-app/app/src/androidTest/java/de/dhbw/ka/se/fibo/login/LoginActivityTest.java:188]: Here you could use a variable (e.g. String password = "testPassword") but I think its not necessary

**Review by Jonas-Karl — DISMISSED** [review]
Everything works fine. I just suggest minor changes in the xml files (match_parent -> wrap_content). Nothing will change in the design but I think its still better to use this.

docs: https://m2.material.io/components/text-fields/android#using-text-fields

**Review by jens1o — APPROVED** [review]
works like a charm, with tests. Great work Markus!

I don't know the reasoning why Jonas is suggesting these and I couldn't find any documentation, but I don't see this as a blocking change. It's up to you two to discuss this.

**Review by Jonas-Karl — APPROVED** [review]

**Pull request merged**